assets/js/base/components/product-grid/container.js

```diff
@@ -56,30 +57,40 @@ ProductGridContainer.propTypes = {
 	urlParameterSuffix: PropTypes.string,
 };
 
-export default withWindow(
-	( { location, history }, { urlParameterSuffix } ) => {
-		const pageParameter = `page${ urlParameterSuffix }`;
-		const orderParameter = `order${ urlParameterSuffix }`;
-		return {
-			currentPage: parseInt( getQueryArg( location.href, pageParameter ) ),
-			orderValue: getQueryArg( location.href, orderParameter ),
-			onPageChange( newPage ) {
-				history.pushState(
-					null,
-					'',
-					addQueryArgs( location.href, { [ pageParameter ]: newPage })
-				)
-			},
-			onOrderChange( orderValue ) {
-				history.pushState(
-					null,
-					'',
-					addQueryArgs( location.href, {
-						[ orderParameter ]: orderValue,
-						[ pageParameter ]: 1,
-					} )
-				);
-			},
-		};
-	}
-)( ProductGridContainer );
+export default withBrowserLocation( ( location, { urlParameterSuffix } ) => {
+	const pageParameter = `page${ urlParameterSuffix }`;
+	const orderParameter = `order${ urlParameterSuffix }`;
+	return {
+		currentPage: parseInt( getQueryArg( location.href, pageParameter ) ),
+		location,
+		orderValue: getQueryArg( location.href, orderParameter ),
+		pageParameter,
+		orderParameter,
+	};
+} )(
+	withBrowserHistory(
+		( history, { location, orderParameter, pageParameter } ) => {
+			return {
+				onPageChange( newPage ) {
+					history.pushState(
+						null,
+						'',
+						addQueryArgs( location.href, {
+							[ pageParameter ]: newPage,
+						} )
+					);
+				},
+				onOrderChange( orderValue ) {
+					history.pushState(
+						null,
+						'',
+						addQueryArgs( location.href, {
+							[ orderParameter ]: orderValue,
+							[ pageParameter ]: 1,
+						} )
+					);
+				},
+			};
+		}
+	)( ProductGridContainer )
+);
```

Comment:
I think it'd be a bit cleaner and easier to follow here if you use `compose` from `@wordpress/element`, or are you trying to avoid use of `compose` for dependency size reasons?

Reply:
Right, we avoid the usage of `@wordpress/element` in frontend scripts for size reasons.